### ISSUE TITLE
Python 2 support

### ISSUE THREAD

**Issue by RomeoDespres** [RomeoDespres/reapy#7]
It would be nice to ensure Python 2.7+ is supported by `reapy`.

**Comment by kostikvento** [comment]
Python2 support is in process, you can check out branch https://github.com/kostikvento/reapy/tree/py2-support, grab `reapy` folder and try using it (at your own risk).

The **only thing** that is **impossible** to support in Python 2 is `reapy.print` (because `print` cannot be a function name — it is a reserved word in Python 2). Use `reapy.show_console_message` or alias `reapy.print_` instead.

So far, it works on my MacOSX Sierra under Python 2.7, and I hope to continue refactoring.

Feel free to share your feedback about it.

**Comment by RomeoDespres** [comment]
So, to be honest I'm a bit worried about this whole Python 2 thing.

On the one hand, for the last 5 to 10 years it's been more and more accepted that Python 2 time was done. Its official end is planned in [6 months from now](https://pythonclock.org/). Many large open source projects have already stated they would stop supporting Python 2 from this date (see list [here](https://python3statement.org/)). It's the main reason why I've not been working on this issue since I've created it: I was not sure it was worth it.

On the other hand, you've already done almost all the work to make `reapy` Python 2-compatible. So why not just integrate it in the code base and be done with it?

But I think our [recent discussion](https://github.com/RomeoDespres/reapy/pull/26#issuecomment-503963745) is a good example of why we might not want to. I'm pretty sure your contribution would have been simpler and more elegant if you hadn't tried to make it Python 2 compatible. Because here is the thing: once we decide to support Python 2, everything from now on must be Python 2 compatible. And this makes usage of the standard library harder: no `ConfigParser`, no `ContextDecorator`... These are only the two examples we've ran into so far, but I'm afraid in the future there may be more and more. And it really feels like a bad idea to me if each time we have to manually reimplement stuff that has been carefully designed in the standard library, for the sake of a dying language.

It's kind of the same with the co-existence of `reapy.utils` and `reapy.tools`. Maybe we could refactor it in a nicer fashion but, once again, I'm worried it's just the first of a long list of future issues.

Maybe I wouldn't have said so two years ago, but now that the end of Python 2 is so close, it seems better to me to keep a clean Python 3 basis.
